src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,177 @@
+package edu.wpi.grip.core;
+
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Point;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+
+/**
+ * Commonly used {@link SocketHint SocketHints}
+ */
+public final class SocketHints {
+
+    /**
+     * Provides some default domains
+     */
+    public enum Domain {
```

Comment:
I'm gonna go with Tom on this one. It adds the potential for confusion without giving enough convenience to make it worth it, IMO.